cmd/unused/internal/ui/group_table.go

```diff
@@ -0,0 +1,64 @@
+package ui
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/grafana/unused"
+)
+
+func GroupTable(ctx context.Context, options Options) error {
+	disks, err := listUnusedDisks(ctx, options.Providers)
+	if err != nil {
+		return err
+	}
+
+	if options.Filter.Key != "" {
+		filtered := make(unused.Disks, 0, len(disks))
+		for _, d := range disks {
+			if d.Meta().Matches(options.Filter.Key, options.Filter.Value) {
+				filtered = append(filtered, d)
+			}
+		}
+		disks = filtered
+	}
+
+	if len(disks) == 0 {
+		fmt.Println("No disks found")
+		return nil
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 8, 4, 2, ' ', 0)
+
+	headers := []string{"PROVIDER", "GROUP_BY", "TYPE", "SIZE_GB"}
```

Comment:
I'm not so sure about the `TYPE` header. When I was looking at the examples I've found it confusing in the second one that there were two entries for both `us-central1-a` and `us-central1-f`, until I figured out it was grouping by `-group-by` **and** by type.

Reply:
Disk type is needed as hdd and ssd storages have different costs and we'd like to know usage for them separately.
As an idea there: we could sort output so records for the same group will be together, for example:
```
PROVIDER  GROUP_BY                         TYPE    COUNT   TOTAL_SIZE_GB
GCP       NONE                             ssd     1       1000
GCP       aerospike                        hdd     3       1125
GCP       agent-smoke-test                 ssd     1       30
GCP       alerting                         ssd     2       40
GCP       billing-prometheus               hdd     1       300
GCP       christian-test                   hdd     1       5
GCP       christian-test                   ssd     1       50
GCP       cortex-chunks                    ssd     7       350
GCP       cortex-dev-01                    hdd     3       300
GCP       cortex-dev-01                    ssd     64      5150
```

what do you think?

Reply:
Yes, the sorting makes it easier to read. Thanks!